eslint.config.mjs

```diff
@@ -0,0 +1,127 @@
+import eslint from '@eslint/js'
+import vitest from '@vitest/eslint-plugin'
+import importPlugin from 'eslint-plugin-import'
+import jestDom from 'eslint-plugin-jest-dom'
+import react from 'eslint-plugin-react'
+import reactCompiler from 'eslint-plugin-react-compiler'
+import reactHooks from 'eslint-plugin-react-hooks'
+import testingLibrary from 'eslint-plugin-testing-library'
+import globals from 'globals'
+import tseslint from 'typescript-eslint'
+
+export default tseslint.config(
+  {
+    ignores: ['**/dist/'],
+  },
+  eslint.configs.recommended,
+  tseslint.configs.recommended,
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+        ...globals.es2015,
+      },
+      parser: tseslint.parser,
+      parserOptions: {
+        ecmaVersion: 2018,
+        sourceType: 'module',
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
+    plugins: {
+      import: importPlugin,
+      'react-compiler': reactCompiler,
+      'react-hooks': reactHooks,
+    },
+    settings: {
+      react: {
+        version: 'detect',
+      },
+      'import/extensions': ['.js', '.jsx', '.ts', '.tsx'],
+      'import/parsers': {
+        '@typescript-eslint/parser': ['.ts', '.tsx'],
+      },
```

Comment:
What would be the default?

Reply:
* default [`import/parsers`](https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#importparsers) is same with our file
* default [`import/extensions`](https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#importextensions) is `js` (but we set react in shared config so default is `js`, `jsx`), but we are using `typescript` without `importPlugin.flatConfigs.typescript`, so we have to add `ts`, `tsx`. I think may be we can combine [`importPlugin.flatConfigs.typescript`](https://github.com/import-js/eslint-plugin-import/blob/main/config/typescript.js) with out custom configs like [this link](https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#typescript). I try it

Reply:
I don't specify the extensions in https://github.com/jotaijs/jotai-valtio/pull/11. I think there must be some defaults we can rely on.

Reply:
> I don't specify the extensions in [jotaijs/jotai-valtio#11](https://github.com/jotaijs/jotai-valtio/pull/11). I think there must be some defaults we can rely on.

@dai-shi I’ll look into it.

Reply:
@dai-shi 

```js
'import/extensions': ['.js', '.jsx', '.ts', '.tsx'],
'import/parsers': {
  '@typescript-eslint/parser': ['.ts', '.tsx'],
},
```

I believe that `import/resolver: { typescript: true }` works as it utilizes `eslint-import-resolver-typescript`, which seems to cover the functionality of the above settings. Therefore, these configurations can likely be omitted while still maintaining functionality. However, this is just a hypothesis and not a confirmed fact. Hence, it’s better to follow the recommendations in this [link](https://github.com/import-js/eslint-plugin-import?tab=readme-ov-file#typescript) when configuring.

Reply:
To be honest, the readme in eslint-plugin-import repo doesn't seem well documented, and I don't trust it.

> However, this is just a hypothesis and not a confirmed fact.

Even if the hypothesis is broken, we only see CI errors, right? Note that our code is basically all TypeScript and can ignore .js in the worst case.

Reply:
@dai-shi I understand, i gonna remove it

Reply:
Thanks. I might be wrong, but I'd like to start minimal and fix it as we find actual issues.

Reply:
@dai-shi Alright, let’s keep an eye on it.